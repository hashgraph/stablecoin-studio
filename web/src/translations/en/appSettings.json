{
	"title": "Configure custom RPC/MIRROR",
	"subtitle": "Choose an option",
	"name": "Name",
	"rpc": "RPC",
	"mirrorNode": "Mirror node",
	"url": "Url",
	"apiKey": "Api Key",
	"addMirror": "Add new service",
	"addRPC": "Add RPC",
	"removeMirror": "Remove Mirror Node",
	"removeRpc": "Remove RPC",
	"header": "Header",
	"mirrorNameExists": "Mirror name already exists, if you want to update it remove it first and recreate it",
	"rpcNameExists": "RPC name already exists, if you want to update it remove it first and recreate it"
}
