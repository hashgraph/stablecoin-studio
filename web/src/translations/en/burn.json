{
    "title": "Burn tokens operation",
    "operationTitle": "Operation details",
    "amountLabel": "Enter the amount",
    "amountPlaceholder": "0",
    "originAccountPlaceholder": "0.0.0",
    "modalSuccessDesc": "{{amount}} tokens have been burned.",
    "modalAction": {
        "subtitle": "Operation details review",
        "originAccount": "Account",
        "amount": "Amount",
        "accept": "Burn tokens"
    }
}