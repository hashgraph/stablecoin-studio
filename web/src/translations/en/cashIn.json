{
    "title": "Cash in operation",
    "operationTitle": "Operation details",
    "amountLabel": "Introduce amount",
    "amountPlaceholder": "0",
    "destinationAccountLabel": "Destination account",
    "destinationAccountPlaceholder": "0.0.0",
    "modalSuccessDesc": "The cash in operation finished successfully.",
    "modalAction": {
        "subtitle": "Operation details review",
        "destinationAccount": "Destination account",
        "amount": "Amount",
        "accept": "Cash in tokens"
    }
}