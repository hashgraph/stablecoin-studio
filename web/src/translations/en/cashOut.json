{
    "title": "Cash out operation",
    "operationTitle": "Operation details",
    "amountLabel": "How much tokens?",
    "amountPlaceholder": "Introduce amount",
    "originAccountLabel": "From which account tokens will be withdrawn?",
    "originAccountPlaceholder": "Introduce account",
    "modalSuccessDesc": "The cash out operation finished successfully.",
    "modalAction": {
        "subtitle": "Operation details review",
        "originAccount": "Account to cash out",
        "amount": "Amount",
        "accept": "Make cash out"
    }
}