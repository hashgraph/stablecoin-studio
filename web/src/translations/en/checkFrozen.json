{
	"title": "Check account frozen status",
	"operationTitle": "Operation details",
	"accountLabel": "Enter the account",
	"accountPlaceholder": "0.0.0",
	"modalIsFrozen": "The account {{ account }} is frozen for the token.",
	"modalIsNotFrozen": "The account {{ account }} is unfrozen for the token.",
	"modalAction": {
		"subtitle": "Operation details review",
		"account": "Account",
		"accept": "Get account frozen status"
	}
}
