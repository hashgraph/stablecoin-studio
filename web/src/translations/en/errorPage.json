{
  "stableCoinNotSelected": {
    "title": "Choose a stable coin to start operating!",
    "description": "You have not chosen a stable coin yet.",
    "description2": "Choose an item to operate."
  }
}