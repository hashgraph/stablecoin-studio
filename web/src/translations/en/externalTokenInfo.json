{
    "common": {
      "nextStep" : "Next step",
      "goBack" : "Go back",
      "createStableCoin": "Create stable coin",
      "createNewStableCoin": "Import token",
      "cancel": "Cancel",
      "addExternalToken": "Import token"
    },
    "tabs": {
      "externalTokenInfo": "Token info"
    },
    "externalTokenInfo": {
      "stableCoinId": "Stable coin id",
      "stableCoinIdPlaceholder": "0.0.0",
      "title": "Imported token",
      "autoCheckRoles": "Do you want to get the roles automatically? This operation will incur a cost",
      "rolesPlaceholder": "Select the roles"
    },
    "optionalDetails": {
      "decimals": "Decimals",
      "expirationDate": "Expiration date",
      "finite": "Finite",
      "initialSupply": "Initial supply",
      "infinite": "Infinite",
      "placeholder": "Enter the {{placeholder}}",
      "maxSupply": "Max supply",
      "title": "Stable coin optional details",
      "typeSupply": "Type supply",
      "typeSupplyPlaceholder": "Choose type supply"
    },
    "managementPermissions": {
      "currentUserKey": "Current user key",
      "description": "Allow all the following features to be managed through smart contracts",
      "pause": "Pause",
      "freeze": "Freeze",
      "introduce": "Enter the {{name}}",
      "none": "None",
      "otherKey": "Other key",
      "placeholder": "Enter the {{placeholder}}",
      "rescue": "Rescue",
      "admin": "Admin",
      "supply": "Supply",
      "theSmartContract": "The smart contract",
      "title": "Stable coin management permissions",
      "totalSupply": "Total supply",
      "treasuryAccountAddress": "Treasury account address",
      "wipe": "Wipe"
    },
    "review": {
      "basicDetails": "Basic details",
      "managementPermissions": "Management permissions",
      "optionalDetails": "Optional details",
      "title": "Stable coin review"
    },
    "notification": {
      "title": "{{ result }} creating the stable coin",
      "descriptionSuccess": "The stable coin was created",
      "descriptionError": "Sorry, there was an error creating the stable coin"
    }
  }