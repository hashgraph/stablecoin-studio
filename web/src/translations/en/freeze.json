{
    "title": "Freeze operation",
    "operationTitle": "Operation details",
    "accountLabel": "Enter the account",
    "accountPlaceholder": "0.0.0",
    "modalSuccess": "The account {{ account }} is frozen.",
    "modalAction": {
        "subtitle": "Operation details review",
        "account": "Account",
        "accept": "Freeze account"
    }
}