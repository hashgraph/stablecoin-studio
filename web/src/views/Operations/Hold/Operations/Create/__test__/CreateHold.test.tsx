import userEvent from '@testing-library/user-event';
import { CreateOperationHold } from '..';
import { render } from '../../../../../../test';
import en from '../../../../../../translations/en/operations.json';
import { waitFor } from '@testing-library/react';

const translations = en;

describe(`<${CreateOperationHold.name} />`, () => {
	test('should render correctly', () => {
		const component = render(<CreateOperationHold />);

		expect(component.asFragment()).toMatchSnapshot();
	});

	test('should render titles', () => {
		const component = render(<CreateOperationHold />);

		expect(component.getByTestId('title')).toHaveTextContent(translations.hold.create.title);
		expect(component.getByTestId('operation-title')).toHaveTextContent(
			translations.hold.create.operationTitle,
		);
	});

	test('should have a disabled confirm button that is enable when introduce valid data', async () => {
		const component = render(<CreateOperationHold />);

		const button = component.getByTestId('confirm-btn');
		expect(button).toBeDisabled();

		const amount = component.getByTestId('amount');
		await userEvent.type(amount, '10000');
		const inputDatePicker = component.getByPlaceholderText(
			translations.hold.create.expirationDatePlaceholder,
		);
		await userEvent.type(inputDatePicker, '01/01/2100');
		const escrow = component.getByTestId('escrow');
		await userEvent.type(escrow, '0.0.1');
		const targetId = component.getByTestId('target');
		await userEvent.type(targetId, '0.0.2');

		await waitFor(() => {
			expect(button).toBeEnabled();
		});
	});

	test('should handle create hold operation', async () => {
		const component = render(<CreateOperationHold />);

		const button = component.getByTestId('confirm-btn');
		expect(button).toBeDisabled();

		const amount = component.getByTestId('amount');
		await userEvent.type(amount, '10000');
		const inputDatePicker = component.getByPlaceholderText(
			translations.hold.create.expirationDatePlaceholder,
		);
		await userEvent.type(inputDatePicker, '01/01/2100');
		const escrow = component.getByTestId('escrow');
		await userEvent.type(escrow, '0.0.1');
		const targetId = component.getByTestId('target');
		await userEvent.type(targetId, '0.0.2');

		await userEvent.click(button);

		const confirmModalButton = component.getByTestId('modal-action-confirm-button');
		await userEvent.click(confirmModalButton);
	});
});
