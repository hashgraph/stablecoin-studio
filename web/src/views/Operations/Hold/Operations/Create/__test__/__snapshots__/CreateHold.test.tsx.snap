// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateOperationHold /> should render correctly 1`] = `
<DocumentFragment>
  <div
    class="chakra-stack css-16oka5y"
  >
    <h2
      class="chakra-heading css-bqsusc"
      data-testid="base-container-heading"
    >
      Operations
    </h2>
    <div
      class="css-1b6lda7"
      data-testid="base-container-children"
    >
      <div
        class="css-1us6gcn"
      >
        <div
          class="css-1r9sqqd"
        >
          <div
            class="chakra-stack css-n21gh5"
          >
            <h2
              class="chakra-heading css-utvjhb"
              data-testid="title"
            >
              Create hold operation
            </h2>
            <p
              class="chakra-text css-1lyqlf8"
              data-testid="operation-title"
            >
              Operation details
            </p>
            <form
              class="chakra-stack css-b5leqj"
            >
              <div
                class="chakra-stack css-1h0sumw"
              >
                <div
                  class="chakra-form-control css-1kxonj9"
                  role="group"
                >
                  <div
                    class="css-gg4vpm"
                  >
                    <label
                      class="chakra-form__label css-1hfoxt1"
                      for="field-:r1:"
                      id="field-:r1:-label"
                    >
                      <div
                        class="chakra-stack css-84zodg"
                      >
                        <p
                          class="chakra-text css-mgupmj"
                        >
                          Enter the amount
                        </p>
                        <p
                          class="chakra-text css-ftb7fv"
                          data-testid="required"
                        >
                          *
                        </p>
                      </div>
                    </label>
                  </div>
                  <div
                    class="chakra-input__group css-bx0blc"
                    data-group="true"
                  >
                    <input
                      autocomplete="off"
                      class="chakra-input css-l1kxkf"
                      data-testid="amount"
                      id="field-:r1:"
                      name="amount"
                      placeholder="0"
                      value=""
                    />
                  </div>
                </div>
              </div>
              <div
                class="chakra-stack css-1h0sumw"
              >
                <div
                  class="chakra-form-control css-1kxonj9"
                  data-testid="form-control"
                  role="group"
                >
                  <label
                    class="chakra-form__label css-1hfoxt1"
                    for="field-:r2:"
                    id="field-:r2:-label"
                  >
                    <div
                      class="chakra-stack css-84zodg"
                    >
                      <p
                        class="chakra-text css-mgupmj"
                      >
                        Expiration date
                      </p>
                      <p
                        class="chakra-text css-x3tq0c"
                      >
                        *
                      </p>
                    </div>
                  </label>
                  <div
                    class="react-datepicker-wrapper"
                  >
                    <div
                      class="react-datepicker__input-container"
                    >
                      <div
                        class="chakra-input__group css-1rp2huo"
                        data-group="true"
                      >
                        <div
                          class="chakra-input__left-element css-nc17it"
                        >
                          <div
                            class="css-1n2mv2k"
                          >
                            <svg
                              class="chakra-icon css-g5uxme"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="0 0 256 256"
                              width="1em"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <rect
                                fill="none"
                                height="256"
                                width="256"
                              />
                              <rect
                                fill="none"
                                height="176"
                                rx="8"
                                stroke="currentColor"
                                stroke-linecap="round"
                                stroke-linejoin="round"
                                stroke-width="16"
                                width="176"
                                x="40"
                                y="40"
                              />
                              <line
                                fill="none"
                                stroke="currentColor"
                                stroke-linecap="round"
                                stroke-linejoin="round"
                                stroke-width="16"
                                x1="176"
                                x2="176"
                                y1="24"
                                y2="56"
                              />
                              <line
                                fill="none"
                                stroke="currentColor"
                                stroke-linecap="round"
                                stroke-linejoin="round"
                                stroke-width="16"
                                x1="80"
                                x2="80"
                                y1="24"
                                y2="56"
                              />
                              <line
                                fill="none"
                                stroke="currentColor"
                                stroke-linecap="round"
                                stroke-linejoin="round"
                                stroke-width="16"
                                x1="40"
                                x2="216"
                                y1="88"
                                y2="88"
                              />
                            </svg>
                          </div>
                        </div>
                        <input
                          class="chakra-input css-1jo7i1r"
                          id="field-:r2:"
                          placeholder="Enter the expiration date"
                          value=""
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="chakra-stack css-1h0sumw"
              >
                <div
                  class="chakra-form-control css-1kxonj9"
                  role="group"
                >
                  <div
                    class="css-gg4vpm"
                  >
                    <label
                      class="chakra-form__label css-1hfoxt1"
                      for="field-:r3:"
                      id="field-:r3:-label"
                    >
                      <div
                        class="chakra-stack css-84zodg"
                      >
                        <p
                          class="chakra-text css-mgupmj"
                        >
                          Escrow address
                        </p>
                        <p
                          class="chakra-text css-ftb7fv"
                          data-testid="required"
                        >
                          *
                        </p>
                      </div>
                    </label>
                  </div>
                  <div
                    class="chakra-input__group css-bx0blc"
                    data-group="true"
                  >
                    <input
                      autocomplete="off"
                      class="chakra-input css-l1kxkf"
                      data-testid="escrow"
                      id="field-:r3:"
                      name="escrow"
                      placeholder="Enter the escrow address"
                      value=""
                    />
                  </div>
                </div>
              </div>
              <div
                class="chakra-stack css-1h0sumw"
              >
                <div
                  class="chakra-form-control css-1kxonj9"
                  role="group"
                >
                  <div
                    class="css-gg4vpm"
                  >
                    <label
                      class="chakra-form__label css-1hfoxt1"
                      for="field-:r4:"
                      id="field-:r4:-label"
                    >
                      <div
                        class="chakra-stack css-84zodg"
                      >
                        <p
                          class="chakra-text css-mgupmj"
                        >
                          Destination address
                        </p>
                        <p
                          class="chakra-text css-ftb7fv"
                          data-testid="required"
                        >
                          *
                        </p>
                      </div>
                    </label>
                  </div>
                  <div
                    class="chakra-input__group css-bx0blc"
                    data-group="true"
                  >
                    <input
                      autocomplete="off"
                      class="chakra-input css-l1kxkf"
                      data-testid="target"
                      id="field-:r4:"
                      name="target"
                      placeholder="Enter the destination address"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </form>
          </div>
          <div
            class="chakra-stack css-eunu1i"
          >
            <div
              class="chakra-stack css-wqnj46"
            >
              <h2
                class="chakra-heading css-1udj1g5"
                data-testid="details-title"
              >
                Stablecoin details
              </h2>
              <div
                class="css-13brihr"
                data-testid="button-box"
              >
                <div
                  class="css-1i64jsd"
                >
                  <p
                    class="chakra-text css-ol2bzi"
                    data-testid="details-review-title"
                  >
                    Basic details
                  </p>
                </div>
                <div
                  class="css-0"
                >
                  <div
                    class="css-0"
                  >
                    <div
                      class="css-gg4vpm"
                      data-testid="details-review-detail-0"
                    >
                      <p
                        class="chakra-text css-9s6br4"
                      >
                        Name
                      </p>
                      <div
                        class="chakra-stack css-1c8ysgd"
                      >
                        <p
                          class="chakra-text css-mgupmj"
                        >
                          TEST_ACCELERATOR_SC
                        </p>
                      </div>
                    </div>
                    <hr
                      aria-orientation="horizontal"
                      class="chakra-divider css-zaaihy"
                      title="divider"
                    />
                  </div>
                  <div
                    class="css-0"
                  >
                    <div
                      class="css-gg4vpm"
                      data-testid="details-review-detail-1"
                    >
                      <p
                        class="chakra-text css-9s6br4"
                      >
                        Symbol
                      </p>
                      <div
                        class="chakra-stack css-1c8ysgd"
                      >
                        <p
                          class="chakra-text css-mgupmj"
                        >
                          TEST
                        </p>
                      </div>
                    </div>
                    <hr
                      aria-orientation="horizontal"
                      class="chakra-divider css-zaaihy"
                      title="divider"
                    />
                  </div>
                  <div
                    class="css-0"
                  >
                    <div
                      class="css-gg4vpm"
                      data-testid="details-review-detail-2"
                    >
                      <p
                        class="chakra-text css-9s6br4"
                      >
                        Decimals
                      </p>
                      <div
                        class="chakra-stack css-1c8ysgd"
                      >
                        <p
                          class="chakra-text css-mgupmj"
                        >
                          6
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="css-13brihr"
                data-testid="button-box"
              >
                <div
                  class="css-1i64jsd"
                >
                  <p
                    class="chakra-text css-ol2bzi"
                    data-testid="details-review-title"
                  >
                    Optional details
                  </p>
                </div>
                <div
                  class="css-0"
                >
                  <div
                    class="css-0"
                  >
                    <div
                      class="css-gg4vpm"
                      data-testid="details-review-detail-0"
                    >
                      <p
                        class="chakra-text css-9s6br4"
                      >
                        Initial supply
                      </p>
                      1000
                    </div>
                    <hr
                      aria-orientation="horizontal"
                      class="chakra-divider css-zaaihy"
                      title="divider"
                    />
                  </div>
                  <div
                    class="css-0"
                  >
                    <div
                      class="css-gg4vpm"
                      data-testid="details-review-detail-1"
                    >
                      <p
                        class="chakra-text css-9s6br4"
                      >
                        Total supply
                      </p>
                      1000
                    </div>
                    <hr
                      aria-orientation="horizontal"
                      class="chakra-divider css-zaaihy"
                      title="divider"
                    />
                  </div>
                  <div
                    class="css-0"
                  >
                    <div
                      class="css-gg4vpm"
                      data-testid="details-review-detail-2"
                    >
                      <p
                        class="chakra-text css-9s6br4"
                      >
                        Max supply
                      </p>
                      1000
                    </div>
                    <hr
                      aria-orientation="horizontal"
                      class="chakra-divider css-zaaihy"
                      title="divider"
                    />
                  </div>
                  <div
                    class="css-0"
                  >
                    <div
                      class="css-gg4vpm"
                      data-testid="details-review-detail-3"
                    >
                      <p
                        class="chakra-text css-9s6br4"
                      >
                        Supply type
                      </p>
                      <div
                        class="chakra-stack css-1c8ysgd"
                      >
                        <p
                          class="chakra-text css-mgupmj"
                        >
                          Finite
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="css-3gu2fl"
        >
          <div
            class="chakra-stack css-1amf41o"
          >
            <button
              class="chakra-button css-1urzhdw"
              data-testid="cancel-btn"
              type="button"
            >
              Go back
            </button>
            <button
              class="chakra-button css-qvkb6r"
              data-testid="confirm-btn"
              disabled=""
              type="button"
            >
              Accept
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
